templates/types/streaming/fastapi/app/api/routers/chat.py

```diff
@@ -19,8 +23,78 @@ class _ChatData(BaseModel):
     messages: List[_Message]
 
 
+class _SourceNodes(BaseModel):
+    id: str
+    metadata: Dict[str, Any]
+    score: Optional[float]
+
+    @classmethod
+    def from_source_node(cls, source_node: NodeWithScore):
+        return cls(
+            id=source_node.node.node_id,
+            metadata=source_node.node.metadata,
+            score=source_node.score,
+        )
+
+    @classmethod
+    def from_source_nodes(cls, source_nodes: List[NodeWithScore]):
+        return [cls.from_source_node(node) for node in source_nodes]
+
+
 class _Result(BaseModel):
     result: _Message
+    nodes: List[_SourceNodes]
+
+
+class VercelStreamResponse(StreamingResponse):
```

Comment:
better move this to a own file

Reply:
probably then `_SourceNodes` too otherwise you'll have a circular import

Reply:
Thank you, I updated by moving the VercelStreamResponse to a new file but still keeping the _SourceNodes and event_generator in this file. Please let me know if there are any issues.